src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/AutoInsert/AutoClosingTagOnAutoInsertProvider.cs

```diff
@@ -39,55 +37,40 @@ internal sealed class AutoClosingTagOnAutoInsertProvider(RazorLSPOptionsMonitor
+    public InsertTextEdit? TryResolveInsertion(Position position, RazorCodeDocument codeDocument, bool enableAutoClosingTags)
     {
-        if (!_optionsMonitor.CurrentValue.AutoClosingTags)
-        {
-            format = default;
-            edit = default;
-            return false;
-        }
-
-        if (!context.SourceText.TryGetAbsoluteIndex(position, out var afterCloseAngleIndex))
-        {
-            format = default;
-            edit = default;
-            return false;
-        }
-
-        if (!TryResolveAutoClosingBehavior(context, afterCloseAngleIndex, out var tagName, out var autoClosingBehavior))
+        if (!(enableAutoClosingTags
+            && codeDocument.Source.Text is { } sourceText
+            && sourceText.TryGetAbsoluteIndex(position, out var afterCloseAngleIndex)
+            && TryResolveAutoClosingBehavior(codeDocument, afterCloseAngleIndex) is { } tagNameWithClosingBehavior))
         {
-            format = default;
-            edit = default;
-            return false;
+            return default;
         }
 
-        if (autoClosingBehavior == AutoClosingBehavior.EndTag)
+        if (tagNameWithClosingBehavior.AutoClosingBehavior == AutoClosingBehavior.EndTag)
         {
-            format = InsertTextFormat.Snippet;
-            edit = VsLspFactory.CreateTextEdit(position, $"$0</{tagName}>");
+            var formatForEndTag = InsertTextFormat.Snippet;
+            var editForEndTag = VsLspFactory.CreateTextEdit(position, $"$0</{tagNameWithClosingBehavior.TagName}>");
 
-            return true;
+            return new InsertTextEdit(editForEndTag, formatForEndTag);
         }
 
-        Debug.Assert(autoClosingBehavior == AutoClosingBehavior.SelfClosing);
+        Debug.Assert(tagNameWithClosingBehavior.AutoClosingBehavior == AutoClosingBehavior.SelfClosing);
 
-        format = InsertTextFormat.Plaintext;
+        var format = InsertTextFormat.Plaintext;
 
         // Need to replace the `>` with ' />$0' or '/>$0' depending on if there's prefixed whitespace.
-        var insertionText = char.IsWhiteSpace(context.SourceText[afterCloseAngleIndex - 2]) ? "/" : " /";
-        edit = VsLspFactory.CreateTextEdit(position.Line, position.Character - 1, insertionText);
+        var insertionText = char.IsWhiteSpace(sourceText[afterCloseAngleIndex - 2]) ? "/" : " /";
+        var edit = VsLspFactory.CreateTextEdit(position.Line, position.Character - 1, insertionText);
 
-        return true;
+        return new InsertTextEdit(edit, format);
     }
 
-    private static bool TryResolveAutoClosingBehavior(FormattingContext context, int afterCloseAngleIndex, [NotNullWhen(true)] out string? name, out AutoClosingBehavior autoClosingBehavior)
+    private static TagNameWithClosingBehavior? TryResolveAutoClosingBehavior(RazorCodeDocument codeDocument, int afterCloseAngleIndex)
```

Comment:
Same honest question - why? :)

Reply:
I don't know if I agree that it will clean up the calling code by a significant amount, and I do have counter-argument above, but since both you and David are in favor of changing back to returning bool, I'm not going to argue. I'll go ahead and change it back.